rdkit/Chem/GraphDescriptors.py

```diff
@@ -330,24 +312,28 @@ def _pyChi0n(mol):
   This makes a big difference after we get out of the first row.
 
   """
-  deltas = [_nVal(x) for x in mol.GetAtoms()]
-  while deltas.count(0):
-    deltas.remove(0)
+  deltas = []
+  for x in mol.GetAtoms():
+    value = _nVal(x)
+    if value != 0:
+      deltas.append(value)
```

Comment:
I don't understand why should it needed to loop twice, while I can make a check for that

Reply:
Either is fine as long it's not the original `while` loop. That looks too inefficient. Going through the list repeatedly instead of having only one pass.